crates/containerd-shim-wasmedge/src/lib.rs

```diff
@@ -8,6 +12,19 @@ pub mod oci_utils;
 #[cfg(unix)]
 pub mod executor;
 
+pub fn parse_version() {
+    let os_args: Vec<_> = env::args_os().collect();
+    let flags = parse(&os_args[1..]).unwrap();
+    if flags.version {
+        println!("{}:", os_args[0].to_string_lossy());
+        println!("  Version: {}", env!("CARGO_PKG_VERSION"));
+        println!("  Revision: {}", env!("CARGO_GIT_HASH"));
+        println!();
+
+        std::process::exit(0);
+    }
+}
```

Comment:
**suggestion:** create a `containerd_shim_wasm::run` function that takes the version / revision, and handles this logic.

Reply:
Sure

Reply:
I don't think there is a good way to handle this in containerd-shim-wasm crate, as each shim implementation still needs to read os args, parse flags and get the version / revision... Could not save much logic here.

Reply:
I meant something like:
```rust
// somewhere in containerd-shim-wasm
pub fn run<Shim: ...>(runtime_id: &str, opts: Option<Config>, version: &str, revision: &str) {
    let os_args: Vec<_> = env::args_os().collect();
    let flags = parse(&os_args[1..]).unwrap();
    if flags.version {
        println!("{}:", os_args[0].to_string_lossy());
        println!("  Version: {}", env!("CARGO_PKG_VERSION"));
        println!("  Revision: {}", env!("CARGO_GIT_HASH"));
        println!();

        std::process::exit(0);
    }
    shim::run::<Shim>(runtime_id, opts)
}
```

and
```rust
// in main.rs
use containerd_shim_wasm::run;
fn main() {
    run::<ShimCli<WasiInstance>>(
        "io.containerd.wasmtime.v1",
        None,
        env!("CARGO_PKG_VERSION"),
        env!("CARGO_GIT_HASH"),
    );
}
```

then you can remove `parse_version` from all the `lib.rs`